students/JasneetChandok/lesson04/assignment/basic_operations.py

```diff
@@ -0,0 +1,131 @@
+logger.setLevel(logging.INFO)
+
+database = SqliteDatabase("customers.db")
+database.connect()
+database.execute_sql("PRAGMA foreign_keys=ON;")
+
+
+class BaseModel(Model):
+    class Meta:
+        database = database
+
+
+class Customer(BaseModel):
+    customer_id = IntegerField(primary_key=True)
+    name = CharField(max_length=50)
+    lastname = CharField(max_length=50)
+    home_address = CharField(max_length=50)
+    phone_number = CharField(max_length=50)
+    email = CharField(max_length=50)
+    status = CharField(max_length=10)
+    credit_limit = IntegerField()
+
+
+def add_customer(customer_id, name, lastname, home_address,
+                 phone_number, email, status, credit_limit):
+    try:
+        with database.transaction():
+            customer = Customer.create(
+                customer_id=customer_id,
+                name=name,
+                lastname=lastname,
+                home_address=home_address,
+                phone_number=phone_number,
+                email=email,
+                status=status,
+                credit_limit=credit_limit,
+            )
+            customer.save()
+        logger.info(f"Customer {customer_id} added to database")
+    except IntegrityError as e:
+        logger.error(f'Customer {customer_id} not added to database: {e}')
+
+
+def search_customer(customer_id):
+    customer_details = {}
+    try:
+        customer = Customer.get(Customer.customer_id == customer_id)
+        customer_details['name'] = customer.name
+        customer_details['lastname'] = customer.lastname
+        customer_details['phone_number'] = customer.phone_number
+        customer_details['email'] = customer.email
+    except CustomerDoesNotExist as e:
+        logger.error(f'Customer not found: customer_id: {customer_id} {e}')
+    finally:
+        return customer_details
+
+
+def delete_customer(customer_id):
+    try:
+        with database.transaction():
+            customer = Customer.get(Customer.customer_id == customer_id)
+            customer.delete_instance()
+            customer.save()
+        logger.info(f"Customer {customer_id} deleted")
+        return True
+    except Exception as e:
+        logger.error(f"Customer {customer_id} not found in database. Skipping delete.")
+        return {}
+
+
+def update_customer_credit(customer_id, credit_limit):
+    try:
+        with database.transaction():
+            customer = Customer.get(Customer.customer_id == customer_id)
+            customer.update(Customer.credit_limit == credit_limit)
+            customer.save()
+        logger.info(f"Credit limit updated for customer {customer_id}")
+    except Exception as e:
+        logger.error(f"Credit limit not updated for customer {customer_id}. "
+                     f"Customer does not exist: {e}")
+        raise ValueError(f"Customer does not exist. {e}")
+
+
+def list_active_customers():
+    query = (Customer.select().where(Customer.status == 'active').count())

```

Comment:
Instead of setting query equal to that and returning it, you can just return it by itself:
```return Customer.select().where(Customer.status == 'active').count()```